### PR TITLE
[JENKINS-52307] fix npe with empty matrix projects

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <olamy@apache.org>

# Description

See [JENKINS-52307](https://issues.jenkins-ci.org/browse/JENKINS-52307).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MatrixProjectImpl.java [hunk 1]
```diff
@@ -72,7 +73,7 @@ public BluePipeline getDynamic(String name) {
 
     @Override
     public BluePipelineContainer getPipelines() {
-        return null;
+        return new PipelineContainerImpl(getOrganization(), this.matrixProject, this);
```
Comment (vivek) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MatrixProjectImpl.java]: We do not support folders in blueocean. Null return is intentional. 🐛

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineFolderImpl.java [hunk 2]
```diff
@@ -186,14 +186,11 @@ public BlueIcon getIcon() {
...
     @Override
     public Iterable<String> getPipelineFolderNames() {
-        return Iterables.transform(getPipelines(), new Function<BluePipeline, String>() {
-            @Override
-            public String apply(@Nullable BluePipeline input) {
-                if (input != null && input instanceof BluePipelineFolder) {
-                    return input.getName();
-                }
-                return null;
+        return Iterables.transform(getPipelines(), input -> {
```
Comment (vivek) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineFolderImpl.java]: NPE was somewhere here? Maybe if we fix it here we should be good?

Comment (olamy) [reply blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineFolderImpl.java]: yup ezy

**Comment by olamy** [comment]
There is a flaky test I will fix that first.

**Comment by vivek** [comment]
@

**Comment by vivek** [comment]
@olamy the artifact ath is unrelated to this pr, I think it’s safe to merge this pr.

**Comment by olamy** [comment]
@vivek k I only have some doubt regarding this failing test https://ci.blueocean.io/blue/organizations/jenkins/blueocean/detail/bugfix%2FJENKINS-52307_multi_job_project_type/5/tests

**Comment by vivek** [comment]
Ok, that’s flaky or legit? let’s get that fixed.

**Pull request merged**